packages/manager/apps/pci-public-ip/src/hooks/useGuideLink/useGuideLink.tsx

```diff
@@ -0,0 +1,35 @@
+import { useContext, useMemo } from 'react';
+import { ShellContext } from '@ovh-ux/manager-react-shell-client';
+import { OvhSubsidiary } from '@ovh-ux/manager-react-components';
+import { GUIDES } from './useGuideLink.constants';
+
+type GetGuideLinkProps = {
+  name?: string;
+  subsidiary: OvhSubsidiary;
+};
+
+function getGuideListLink({ subsidiary }: GetGuideLinkProps) {
+  return Object.entries(GUIDES).reduce(
+    (result, [key, value]) => ({
+      ...result,
+      [key]: value[subsidiary] ?? value.DEFAULT,
+    }),
+    {} as { [guideName: string]: string },
+  );
+}
```

Comment:
There is no need to type cast the initial value, you can add type as generic argument of reduce function :)
(BTW, you can use UseGuideLinkProps)
```suggestion
function getGuideListLink({ subsidiary }: GetGuideLinkProps) {
  return Object.entries(GUIDES).reduce<UseGuideLinkProps>(
    (result, [key, value]) => ({
      ...result,
      [key]: value[subsidiary] ?? value.DEFAULT,
    }),
    {},
  );
}
```